bundles/org.eclipse.swt/Eclipse SWT/common/org/eclipse/swt/graphics/ImageLoader.java

```diff
@@ -155,14 +155,19 @@ public ImageData[] load(InputStream stream) {
 	return data;
 }
 
-List<ElementAtZoom<ImageData>> load(InputStream stream, int fileZoom, int targetZoom) {
+List<ImageData> load(InputStream stream, int fileZoom, int targetZoom) {
 	if (stream == null) SWT.error(SWT.ERROR_NULL_ARGUMENT);
 	reset();
-	List<ElementAtZoom<ImageData>> images = NativeImageLoader.load(new ElementAtZoom<>(stream, fileZoom), this, targetZoom);
-	data = images.stream().map(ElementAtZoom::element).toArray(ImageData[]::new);
+	List<ImageData> images = NativeImageLoader.load(new ElementAtZoom<>(stream, fileZoom), this, targetZoom);
+	data = images.stream().toArray(ImageData[]::new);
 	return images;
 }
 
+boolean canLoadAtZoom(InputStream stream, int fileZoom, int targetZoom) {
+	if (stream == null) SWT.error(SWT.ERROR_NULL_ARGUMENT);
+	return NativeImageLoader.canLoadAtZoom(new ElementAtZoom<>(stream, fileZoom), targetZoom);
```

Comment:
Calling `FileFormat.canLoadAtZoom()` here directly would save us the indirection via `NativeImageLoader` that is identical over all three impls.
```suggestion
	return FileFormat.canLoadAtZoom(new ElementAtZoom<>(stream, fileZoom), targetZoom);
```

Reply:
That's true. I added the method there because we wanted to use methods in the native loader from Platform UI as a first implementation, but we can also directly access the FileFormat implementation. I adapted that in the latest push.